Client/Frontend/Home/TopSites/TopSitesDimension.swift

```diff
@@ -12,16 +12,19 @@ struct TopSitesSectionDimension {
 struct TopSitesUIInterface {
     var isLandscape: Bool = UIWindow.isLandscape
     var isIphone: Bool = UIDevice.current.userInterfaceIdiom == .phone
-    var horizontalSizeClass: UIUserInterfaceSizeClass
+    var trait: UITraitCollection
+    var availableWidth: CGFloat
 
-    init(trait: UITraitCollection) {
-        horizontalSizeClass = trait.horizontalSizeClass
+    init(trait: UITraitCollection, availableWidth: CGFloat) {
+        self.trait = trait
+        self.availableWidth = availableWidth
     }
 
-    init(isLandscape: Bool, isIphone: Bool, trait: UITraitCollection) {
+    init(isLandscape: Bool, isIphone: Bool, trait: UITraitCollection, availableWidth: CGFloat) {
```

Comment:
I think I prefer having only one init with default params rather than 2, and having one only used for unit tests. What do you think?